dash/include/dash/matrix/internal/Matrix-inl.h

```diff
@@ -80,13 +80,54 @@ ::Matrix(
 
 template <typename T, dim_t NumDim, typename IndexT, class PatternT>
 inline Matrix<T, NumDim, IndexT, PatternT>
+::Matrix(
+  self_t && other)
+: _team(other._team),
+  _size(other._size),
+  _lcapacity(other._lcapacity),
+  _begin(other._begin),
+  _pattern(other._pattern),
+  _glob_mem(other._glob_mem),
+  _lbegin(other._lbegin),
+  _lend(other._lend),
+  _ref(other._ref)
+{
+    // do not free other globmem
+    other._glob_mem = nullptr;
```

Comment:
... in general: To determine which pointers need to be nulled, look at the destructor. Only the destructor of a moved instance may be called after moving, so dangling is not an issue if the destructor does not depend on the pointers.